### DISCUSSION TITLE
How can I attach data for getUser() to return?

### DISCUSSION THREAD

**Discussion by k2xl** [supabase/supabase#30427] [General]
I have a table called profiles and there is a foreign key from the auth user id to that table.

When I call getUser() i'd like one of the columns to be added to the response to getUser(). Right now I'm having to do it in two separate queries which is inefficient. How can I join the columns I want and have it part of the getUser() response?

**Comment by GaryAustin1** [comment]
Maybe write your own Postgres function to get data from the auth table and the profile table.  Using the user_id from auth.uid() would confirm the jwt if valid and a security definer function could fetch all the data you want.

**Reply by k2xl** [reply]
Is there anything additional that getUser() does that may make rolling my own more complex? Reading the docs it makes me a bit more hesitant (don't want to mess with any cookies/refreshing sessions etc) I'm thinking this is a common use case (i.e. fetch the user but also get any user roles for the application to gatekeep specific actions)

**Reply by GaryAustin1** [reply]
I don't believe it refreshes the session.  The docs do not show session returned or mention any other action.

The code though does seem to wipe your current session if it gets an error.  https://github.com/supabase/auth-js/blob/67d3e865101f8e9b7c8e5332c622722c67f10d33/src/GoTrueClient.ts#L1158 
So indeed it adds a bit of security in that if the user is signed out it would return an error, versus a database call which relies just on the jwt.
